paasta_tools/cli/cmds/spark_run.py

```diff
@@ -404,6 +418,7 @@ def load_aws_credentials_from_yaml(yaml_file_path):
         return (
             credentials_yaml['aws_access_key_id'],
             credentials_yaml['aws_secret_access_key'],
+            credentials_yaml['region'],
```

Comment:
Most of the yaml doesn't have region, maybe use `.get('region')` to ensure it won't break?

Reply:
Looks like order of precedence is:
1. --aws-region
2. region in yaml files (if exists)
3. Session.region_name

Whenever returning from get_aws_credentials_and_region(), we should have a valid region instead of None.

Reply:
> Most of the yaml doesn't have region, maybe use `.get('region')` to ensure it won't break?

ack!

> Looks like order of precedence is:
> 
> 1. --aws-region
> 2. region in yaml files (if exists)
> 3. Session.region_name
> 
> Whenever returning from get_aws_credentials_and_region(), we should have a valid region instead of None.

we could have a variable  DEFAULT AWS REGION at the top of the file, and set it to be 'us-west-2'. Then, we can ensure that get_aws_credentials_and_region() always returns a region.

Reply:
@lydian @huadongliu I think it makes sense not to look for the aws region in the credentials files, since the region really shouldn't be there. 

We can just look if --aws-region is supplied and if not, we can default to 'us-west-2'? This also keeps our logic simple.